newrelic-agent/src/main/java/com/newrelic/agent/service/module/JarCollectorServiceProcessor.java

```diff
@@ -55,32 +58,29 @@
                     Attributes.Name.IMPLEMENTATION_VENDOR.toString(),
                     Attributes.Name.IMPLEMENTATION_VENDOR_ID.toString());
 
-    private final boolean skipTempJars; // default true
     private final Logger logger;
-
-    /**
-     * The list of jars to ignore.
-     */
+    private final boolean skipTempJars;
     private final List<String> ignoreJars;
+    private final RateLimiter processUrlRateLimiter;
 
     public JarCollectorServiceProcessor(Logger logger, AgentConfig agentConfig) {
-        this(agentConfig, agentConfig.getIgnoreJars(), logger);
-    }
-
-    /**
-     * Creates this JarCollectorServiceProcessor.
-     */
-    JarCollectorServiceProcessor(Config config, List<String> ignoreJars, Logger logger) {
-        this.ignoreJars = new ArrayList<>(ignoreJars);
         this.logger = logger;
-        this.skipTempJars = config.getValue("jar_collector.skip_temp_jars", true);
+        this.skipTempJars = agentConfig.getJarCollectorConfig().skipTempJars();
         if (!skipTempJars) {
             logger.log(Level.FINEST, "temporary jars will be transmitted to the host");
         }
+        this.ignoreJars = new ArrayList<>(agentConfig.getIgnoreJars());
+        int jarsPerSecond = agentConfig.getJarCollectorConfig().getJarsPerSecond();
+        if (jarsPerSecond <= 0) {
+            logger.log(Level.INFO, "Jars per second must be greater than 0. Defaulting to {0}.", DEFAULT_JARS_PER_SECOND);
```

Comment:
Should we disable if 0 and use the default if negative?

Reply:
I could do that. I was thinking through that question myself and came to the conclusion that you could effectively disable by setting the config to really high number. Like, if someone set it to 1_000_000_000, there's no way the single thread processing these jars could run into that bound.

Reply:
I'm ok with leaving this behavior as is and providing public documentation for the `jar_collector` config so that customers can disable it if desired.